catalog/dags/common/ingestion_server.py

```diff
@@ -125,3 +142,36 @@ def trigger_and_wait_for_task(
     waiter = wait_for_task(action, trigger, timeout, poke_interval)
     trigger >> waiter
     return trigger, waiter
+
+
+def api_health_check(
+    media_type: str,
+    index_suffix: str,
+    timeout: timedelta,
+    poke_interval: int = POKE_INTERVAL,
+    retries: int = 8,
+) -> HttpSensor:
+    """
+    Poll the API for the given media endpoint, targeting the given index, and return
+    successfully when healthy results are returned.
+
+    If no results are returned, it will reschedule and poll again. If a 400 is received
+    because the API is not able to connect to the given index, the task will fail and
+    retry. The task is set to retry with exponential backoff, such that the retry delay
+    doubles between each attempt.
+    """
+    access_token = Variable.get("API_ACCESS_TOKEN", "not_set")
+    return HttpSensor(
+        task_id="api_health_check",
+        http_conn_id=API_CONN_ID,
+        endpoint=f"{media_type}",
+        request_params={"internal__index": f"{media_type}-{index_suffix}"},
+        headers={"Authorization": f"Bearer {access_token}"},
```

Comment:
The `headers` param of the HTTP sensor is templated, so we should be able to just use the value directly!

```suggestion
    return HttpSensor(
        task_id="api_health_check",
        http_conn_id=API_CONN_ID,
        endpoint=f"{media_type}",
        request_params={"internal__index": f"{media_type}-{index_suffix}"},
        headers={"Authorization": "Bearer {{ var.value.API_ACCESS_TOKEN }}"},
```